src/devices/Nmea0183/SentenceCache.cs

```diff
@@ -92,35 +149,54 @@ public bool TryGetCurrentPosition(out GeographicPosition? position, out Angle tr
         /// GNSS receivers only deliver the position at 1Hz or less)
         /// </summary>
         /// <param name="position">Current position</param>
+        /// <param name="source">Only look at this source (otherwise, if multiple sources provide a position, any is used)</param>
         /// <param name="extrapolate">True to extrapolate the current position using speed and track</param>
         /// <param name="track">Track (course over ground)</param>
         /// <param name="sog">Speed over ground</param>
         /// <param name="heading">Vessel Heading</param>
+        /// <param name="messageTime">Time of the position report that was used</param>
+        /// <param name="now">The current time (when working with data in the past, this may be the a time within that data set)</param>
         /// <returns>True if a valid position is returned</returns>
-        public bool TryGetCurrentPosition(out GeographicPosition? position, bool extrapolate, out Angle track, out Speed sog, out Angle? heading)
+        public bool TryGetCurrentPosition(
+#if NET5_0_OR_GREATER
+            [NotNullWhen(true)]
+#endif
+            out GeographicPosition? position,
+            String? source, bool extrapolate, out Angle track, out Speed sog, out Angle? heading, out DateTimeOffset messageTime, DateTimeOffset now)
         {
+            messageTime = default;
             // Try to get any of the position messages
-            var gll = (PositionFastUpdate?)GetLastSentence(PositionFastUpdate.Id);
-            var gga = (GlobalPositioningSystemFixData?)GetLastSentence(GlobalPositioningSystemFixData.Id);
-            var rmc = (RecommendedMinimumNavigationInformation?)GetLastSentence(RecommendedMinimumNavigationInformation.Id);
-            var vtg = (TrackMadeGood?)GetLastSentence(TrackMadeGood.Id);
-            var hdt = (HeadingTrue?)GetLastSentence(HeadingTrue.Id);
+            var gll = (PositionFastUpdate?)GetLastSentence(source, PositionFastUpdate.Id);
+            var gga = (GlobalPositioningSystemFixData?)GetLastSentence(source, GlobalPositioningSystemFixData.Id);
+            var rmc = (RecommendedMinimumNavigationInformation?)GetLastSentence(source, RecommendedMinimumNavigationInformation.Id);
+            var vtg = (TrackMadeGood?)GetLastSentence(source, TrackMadeGood.Id);
+            var hdt = (HeadingTrue?)GetLastSentence(source, HeadingTrue.Id);
```

Comment:
nit: given this code is complex enough I'd generally not recommend using abbreviations (feel free to ignore in this PR)